AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Sync/MutationSync/OutgoingMutationQueue/OutgoingMutationQueue.swift

```diff
@@ -138,7 +138,8 @@ final class OutgoingMutationQueue: OutgoingMutationQueueBehavior {
                        mutationEventPublisher: MutationEventPublisher) {
         log.verbose(#function)
         self.api = api
-        operationQueue.isSuspended = false
+
+        checkMutationEventDatabaseStatus()
 
         // State machine notification to ".receivedSubscription" will be handled in `receive(subscription:)`
         mutationEventPublisher.publisher.subscribe(self)
```

Comment:
why isn't this moved as well? you can expose a completion for `checkMutationEventDatabaseStatus()` like
```
checkMutationEventDatabaseStatus(onComplete: {
   operationQueue.isSuspended = false
   // State machine notification to ".receivedSubscription" will be handled in `receive(subscription:)`
    mutationEventPublisher.publisher.subscribe(self)
}
```